lib/lints/double_literal_format/double_literal_format_fix.dart

```diff
@@ -0,0 +1,64 @@
+part of 'double_literal_format_rule.dart';
+
+/// A Quick fix for `double-literal-format` rule
+/// Suggests the correct value for an issue
+class _DoubleLiteralFormatFix extends DartFix {
+  @override
+  void run(
+    CustomLintResolver resolver,
+    ChangeReporter reporter,
+    CustomLintContext context,
+    AnalysisError analysisError,
+    List<AnalysisError> others,
+  ) {
+    context.registry.addDoubleLiteral((node) {
+      // checks that the literal declaration is where our warning is located
+      if (!analysisError.sourceRange.intersects(node.sourceRange)) return;
+
+      final lexeme = node.literal.lexeme;
+      String? correctLexeme;
+
+      if (lexeme.hasLeadingZero) {
+        correctLexeme = _correctLeadingZeroLexeme(lexeme);
+      } else if (lexeme.hasLeadingDecimalPoint) {
+        correctLexeme = _correctLeadingDecimalPointLexeme(lexeme);
+      } else if (lexeme.hasTrailingZero) {
+        correctLexeme = _correctTrailingZeroLexeme(lexeme);
+      }
```

Comment:
Can we simplify?
```suggestion
      final correctLexeme = double.parse(lexeme).toString();
```

Reply:
Unfortunately, not. 
This code

```
final valStr = '.16e+5';
final valStr2 = double.parse(valStr).toString();
print('$valStr --- $valStr2');
```
produces output:

.16e+5 --- 16000

i.e. the original lexeme may be modified.